src/scripts/main.js

```diff
@@ -1 +1,67 @@
 'use strict';
+
+const body = document.querySelector('body');
+
+new Promise((resolve, reject) => {
+  body.addEventListener('click', () => {
+    resolve('First promise was resolved');
+  });
+
+  setTimeout(() => {
+    reject(new Error('First promise was rejected'));
+  }, 3000);
+}).then(data => {
+  successHandler(data);
+}).catch(error => {
+  errorHandler(error);
+});
+
+new Promise((resolve, reject) => {
+  body.addEventListener('click', () => {
+    resolve('Second promise was resolved');
+  });
+
+  body.addEventListener('contextmenu', () => {
+    resolve('Second promise was resolved');
+  });
+}).then(data => successHandler(data));
+
+new Promise((resolve, reject) => {
+  // eslint-disable-next-line no-shadow
+  const p1 = new Promise((resolve, reject) => {
+    body.addEventListener('click', () => {
+      resolve('left click');
+    });
+  });
+
+  // eslint-disable-next-line no-shadow
+  const p2 = new Promise((resolve, reject) => {
+    body.addEventListener('contextmenu', () => {
+      resolve('right click');
+    });
+  });
```

Comment:
use 'mousedown' event here to manage left-right click

Reply:
done,  as I understand it